mlir/test/Dialect/Vector/canonicalize/vector-transpose.mlir

```diff
@@ -0,0 +1,114 @@
+// RUN: mlir-opt %s -canonicalize="test-convergence" -split-input-file -allow-unregistered-dialect | FileCheck %s
+
+// This file contains some canonicalizations tests involving vector.transpose.
```

Comment:
Note, it's totally valid (and something I personally encourage) to document what pattern specifically is being tested:
* https://mlir.llvm.org/getting_started/TestingGuide/#making-tests-self-documenting